src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```diff
@@ -249,12 +253,78 @@ public function createAction(Request $request)
+                            $this->trans('This page has been customized in some shops.', 'Admin.Cms.Notification') . ' ' .
+                            $this->trans('Please edit this page in a specific shop.', 'Admin.Cms.Notification')
+                        );
+                        $forceEditUrl = $this->generateUrl('admin_cms_pages_edit', [
+                            'cmsPageId' => $cmsPageId,
+                            'forceEdit' => true,
+                        ]);
+                        $forceEditMessage = $this->trans('Replace all shop specific content with content from shop 1.', 'Admin.Cms.Notification');
+                    }
+                }
+                if ($shopContext == Shop::CONTEXT_SHOP) {
+                    if ($isSame['is_same'] && !$request->request->has('forceEdit')) {
+                        if (!$this->employeeHasAccessAssociatedShops($cmsPageId)) {
+                            throw new CannotEditCmsPageException('You don\'t have permissions to edit this page in all shops it belongs to.', CannotEditCmsPageException::INSUFFICIENT_PERMISSION_ALL_SHOPS);
+                        }
+                        $this->addFlash(
+                            'error',
+                            $this->trans('This page is the same in all shops.', 'Admin.Cms.Notification') . ' ' .
+                            $this->trans('Please edit this page in \'All shops\' context.', 'Admin.Cms.Notification')
+                        );
+                        $forceEditUrl = $this->generateUrl('admin_cms_pages_edit', [
+                            'cmsPageId' => $cmsPageId,
+                            'forceEdit' => true,
+                        ]);
+                        $forceEditMessage = $this->trans('Edit the content only for this shop.', 'Admin.Cms.Notification');
+                    }
+                }
+            }
             /** @var EditableCmsPage $editableCmsPage */
             $editableCmsPage = $this->getQueryBus()->handle(new GetCmsPageForEditing($cmsPageId));
             $previewUrl = $editableCmsPage->getPreviewUrl();
 
+            $submittedData = $request->request->get('cms_page');
+            $addedAssociations = [];
+            if (!empty($submittedData)) {
+                $newShopAssociations = $submittedData['shop_association'] ?? null;
+                if ($newShopAssociations) {
+                    $oldShopAssociations = $editableCmsPage->getShopAssociation();
+                    $employeeAssociations = array_column(
+                        $db->executeS('select id_shop from ' . $prefix . 'employee_shop where id_employee = ' . $context->employee->id),
+                        'id_shop');
+                    $addedAssociations = array_diff($newShopAssociations, $oldShopAssociations);
+                    $removedAssociations = array_diff($oldShopAssociations, $newShopAssociations);
+                    var_dump($employeeAssociations, $addedAssociations, $removedAssociations);
```

Comment:
```suggestion
```

Reply:
Sorry for that. This line escaped my attention...